torch/testing/_internal/common_utils.py

```diff
@@ -3230,17 +3243,21 @@ def _run_custom(self, result=None):
         )
 
         with unittest.mock.patch("torch._dynamo.config.suppress_errors", suppress_errors), maybe_disable_size_asserts:
+            method = getattr(self, self._testMethodName)
             if TEST_WITH_AOT_EAGER:
-                super_run = torch._dynamo.optimize("aot_eager_decomp_partition")(super_run)
+                method = torch._dynamo.optimize("aot_eager_decomp_partition")(method)
+                setattr(self, self._testMethodName, method)
             elif TEST_WITH_TORCHDYNAMO or TEST_WITH_TORCHINDUCTOR:
                 if TEST_WITH_TORCHINDUCTOR:
-                    super_run = torch._dynamo.optimize("inductor")(super_run)
+                    method = torch._dynamo.optimize("inductor")(method)
+                    setattr(self, self._testMethodName, method)
                 else:
                     # Assume eager-generated GraphModules will not error out.
                     # If we do, this is probably a Dynamo bug!
-                    super_run = torch._dynamo.optimize("eager_noexcept", nopython=nopython)(super_run)
+                    method = torch._dynamo.optimize("eager_noexcept", nopython=nopython)(method)
+                    setattr(self, self._testMethodName, method)
 
-                key = f"{self.__class__.__name__}.{self._testMethodName}"
+                key = self._dynamo_test_key()
```

Comment:
I would recommend splitting up this PR a little.
1) all the non-cpython tests
2) adding the cpython test files (this might be one-at-a-time, or in a batch). More of a question for @albanD I think

Also, separately, Alban, Do you think there's a better way for us to reuse the cpython tests instead of copy-pasting and modifying them into pytorch?

Reply:
The changes involve fixing some imports and replacing `unittest.TestCase` with PyTorch's `TestCase` class.